tests/tests.cpp

```diff
@@ -56,6 +61,27 @@ size_t checksize(size_t& n, size_t added)
     return n;
 }
 
+TEST(MegaApi, getMimeType)
+{
+
+    vector<unique_ptr<thread>> threads;
+
+    atomic<int> successCount = 0;
+
+    for (int i = 0; i < 20; ++i)
+    {
+        threads.emplace_back(unique_ptr<thread>(new thread([&successCount]() {
```

Comment:
I find that proving that something's thread-safe is really hard and takes too much effort. Even if this test passes it doesn't mean the function is thread-safe. I would simply test the function in the single-threaded case and ensure all branches are covered. I think these cases are missing:
- input is a nullptr
- the input begins with '.'

Reply:
I've added those two now, thanks.   I think it's important to test the threaded case, since the whole change's purpose is to fix the thread safety, otherwise we may as well not make the change at all.